rosetta-source/src/main/rosetta/product-template-type.rosetta

```diff
@@ -217,15 +217,33 @@ type ReturnTerms: <"Specifies the type of return of a performance payout.">
     [metadata key]
 
     observationTerms ObservationTerms (0..1) <"Defines how and when a performance type option or performance type swap is to be observed.">
     valuationDates ValuationDates (1..1) <"Defines how and when a performance type option or performance type swap is to be valued, including both interim and final valuation.">
     paymentDates PaymentDates (1..1) <"Defines the payment date schedule, as defined by the parameters that are needed to specify it, either in a parametric way or by reference to another schedule of dates (e.g. the valuation dates).">
     underlier Product (0..1) <"Identifies the underlying product that is referenced for pricing of the applicable leg in a swap.  Referenced in the '2018 ISDA CDM Equity Confirmation for Security Equity Swap' as Security.">
     fxFeature FxFeature (0..*) <"Defines quanto or composite FX features that are included in the swap leg.">
-    returnTerms ReturnTerms (1..1) <"Specifies the type of return of a performance payout.">
+    returnTerms ReturnTerms (0..1) <"Specifies the type of return of a performance payout.">
+    portfolioReturnTerms PortfolioReturnTerms (0..*) <"Specifies an individual type of return of a Performance Payout, when such individual return is part of an aggregation of multiple similar returns, at Performance Payout level">
+    initialValuationPrice PriceSchedule (0..1) <"Specifies the net initial valuation price(s) of the underlier at Performance Payout level. This price can be expressed either as an actual amount/currency, as a determination method, or by reference to another value specified in the swap document.">
+    [metadata address "pointsTo"=PriceQuantity->price]
+    interimValuationPrice PriceSchedule (0..1) <"Specifies the net initial valuation price(s) of the underlier at Performance Payout level. This price can be expressed either as an actual amount/currency, as a determination method, or by reference to another value specified in the swap document.">
+    [metadata address "pointsTo"=PriceQuantity->price]
+    finalValuationPrice PriceSchedule (0..1) <"Specifies the net final valuation price(s) of the underlier at Performance Payout level. This price can be expressed either as an actual amount/currency, as a determination method, or by reference to another value specified in the swap document.">
+    [metadata address "pointsTo"=PriceQuantity->price]
+
+    condition PorfolioOrStraightReturn:
+    required choice returnTerms, portfolioReturnTerms
+
+    condition PortofolioReturnIsMultipleReturns :
+    if portfolioReturnTerms exists
+    then portfolioReturnTerms count > 1
+
+    condition UnderlierOfPortfolioIsBasket:
```

Comment:
In this case, the condition description should address the question of why it only applies in certain cases (`priceReturnterms`).

Reply:
do not agree to document all business cases "why" condition exists, i mean that is theoreticall exercise... and to what extent in practice : shall i re-explain all what is about various strategies etc. ? you would potentially require here long descriptions... what would be your suggestion, for instance ?